### PR TITLE
backport: bitcoin#19528, #19455, #19717, #19849, #19994 - assert for RPCArg names

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
This batch of backports asserts that RPCArg names are equal to CRPCCommand ones.

## What was done?
done backports:
 - bitcoin/bitcoin#19994
 - bitcoin/bitcoin#19849
 - bitcoin/bitcoin#19717
 - bitcoin/bitcoin#19455
 - bitcoin/bitcoin#19528
 
Beside that same changes are applied for src/coinjoin's rpc.

There's also applied multiple fixes for various rpcs for cases when RPCArg names are mismatched with CPCCommand

**Please, note, that this PR is not final fix for all RPCArgs**. There's a lot of dash's rpc that is not refactored that.
That it is not easy to implement for `quorum command` because the list of arguments (and even their numbers) are different for each sub-command. This fixes are out-of scope of this PR and should be done before bitcoin#18531 is backported. 

See also relevant bitcoin#21035.

## How Has This Been Tested?
I used this helper to see which exactly args are specified wrongly:
```cpp
diff --git a/src/rpc/server.h b/src/rpc/server.h
index d4a7ba60eb..cdfd741f54 100644
--- a/src/rpc/server.h
+++ b/src/rpc/server.h
@@ -16,6 +16,7 @@
 #include <string>
 
 #include <univalue.h>
+#include <logging.h>
 
 class CRPCCommand;
 
@@ -110,6 +111,19 @@ public:
               fn().GetArgNames(),
               intptr_t(fn))
     {
+        if (fn().m_name != name_in || fn().GetArgNames() != args_in) {
+            std::cerr << "names:  " << fn().m_name << ' ' << name_in << std::endl;
+            std::cerr << "arg names: " << fn().GetArgNames().size() << std::endl;
+            for (const auto& i : fn().GetArgNames()) {
+                std::cerr << "arg: " << i << std::endl;
+            }
+            std::cerr << "FIASCO FIASCO FIASCO FIASCO FIASCO FIASCO" << std::endl;
+        }
         CHECK_NONFATAL(fn().m_name == name_in);
         CHECK_NONFATAL(fn().GetArgNames() == args_in);
     }
```

## Breaking Changes
N/A

Some arguments are renamed in RPC but they have been broken (used incorrect name not same as in docs)

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: test/functional/rpc_rawtransaction.py [hunk 1]
```diff
@@ -339,7 +339,7 @@ def run_test(self):
 
         # 5. valid parameters - supply txid and True for non-verbose
         assert_equal(self.nodes[0].getrawtransaction(txId, True)["hex"], rawTxSigned['hex'])
-        assert_equal(self.nodes[0].getrawtransactionmulti({"0":[txId]}, True)[txId]['hex'], rawTxSigned['hex'])
+        assert_equal(self.nodes[0].getrawtransactionmulti(verbose=True, transactions={"0":[txId]})[txId]['hex'], rawTxSigned['hex'])
```
Comment (UdjinM6) [inline test/functional/rpc_rawtransaction.py:342]: 0009c5e81eb0e07b930eea5bf40bf95ac260e20c: nit: why changing args order here?

**Review by UdjinM6 — APPROVED** [review]
utACK

File: test/functional/rpc_rawtransaction.py (hunk 1 above)
Comment (knst) [reply test/functional/rpc_rawtransaction.py:342]: doesn't really matter, I just wanted to be sure that it is parsed by name not by its number.

Should I revert it?

Comment (UdjinM6) [reply test/functional/rpc_rawtransaction.py:342]: no, that's fine, I was just wondering if I maybe missed something here

**Review by UdjinM6 — DISMISSED** [review]
utACK

**Comment by PastaPastaPasta** [comment]
@knst I rebased this for you; range-diff provided below; feel free to no-diff rebase to re-enshrine your GPG sigs on these commits if you'd like

```
 -:  ---------- >  1:  f5642281cc Merge #20282: wallet: change upgradewallet return type to be an object
 -:  ---------- >  2:  19aba38cab Merge #19200: rpc: remove deprecated getaddressinfo fields
 -:  ---------- >  3:  9c54cb16de Merge #19405: rpc, cli: add network in/out connections to `getnetworkinfo` and `-getinfo`
 1:  6353aae4e9 =  4:  58d923cd5b Merge #19528: rpc: Assert that RPCArg names are equal to CRPCCommand ones (misc)
 2:  8044dd4c91 =  5:  41c35fd8dc fix: adjust missing arguments and help for misc rpc: debug, echo, mnsync
 3:  9950451b37 =  6:  860d31f504 Merge #19455: rpc generate: print useful help and error message
 4:  f99734e3e3 =  7:  7ac1ee0fb4 Merge #19717: rpc: Assert that RPCArg names are equal to CRPCCommand ones (mining,zmq,rpcdump)
 5:  c0439e91ad =  8:  f525f574b0 fix: follow-up missing changes from Merge #18607: rpc: Fix named arguments in documentation
 6:  d631b7a9ba =  9:  c30c8f22dd Merge #19849: Assert that RPCArg names are equal to CRPCCommand ones (blockchain,rawtransaction)
 7:  a827d5f357 = 10:  af9eb81e56 fix: wrong name of arguments for RPC
 8:  36eb6f6082 ! 11:  0e1a31159f Merge #19994: Assert that RPCArg names are equal to CRPCCommand ones (net, rpcwallet)
    @@ src/wallet/rpcwallet.cpp: static UniValue upgradewallet(const JSONRPCRequest& re
          if (!wallet) return NullUniValue;
          CWallet* const pwallet = wallet.get();
     @@ src/wallet/rpcwallet.cpp: static UniValue upgradewallet(const JSONRPCRequest& request)
    -         throw JSONRPCError(RPC_WALLET_ERROR, error.original);
    +         obj.pushKV("error", error.original);
          }
    -     return error.original;
    +     return obj;
     +},
     +    };
      }
 9:  106d69a45f = 12:  d0163543d9 refactor: use new format CPCCommand for rpc/coinjoin
10:  efcbc91de1 = 13:  b54f03a0c1 fix: wrong name of argument for coinjoin
11:  6ae994ba36 = 14:  3621966f12 feat: add todo to drop Throw() from rpc/util.h
12:  1f3d7c6b1c = 15:  c5031685bc fix: rename arguments for 'voteraw'
```

**Review by PastaPastaPasta — APPROVED** [review]
(self)utACK c5031685bce0a64dda7a4c9487414725b5c7f97c

**Review by knst — COMMENTED** [review]
Rebase LGTM c5031685bce0a64dda7a4c9487414725b5c7f97c

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**